tests/subsys/fs/fs_api/src/main.c

```diff
@@ -65,35 +45,30 @@ static void fs_teardown(void)
  * @ingroup filesystem
  *
  */
-void test_main(void)
+ZTEST(fat_fs_basic, test_fat_fs)
 {
-	ztest_test_suite(fat_fs_basic_test,
-			 ztest_unit_test(test_fs_register),
-			 ztest_unit_test_setup_teardown(test_mount,
-							fs_setup,
-							dummy_teardown),
-			 ztest_unit_test(test_fs_file_t_init),
-			 ztest_unit_test(test_fs_dir_t_init),
-			 ztest_unit_test(test_file_statvfs),
-			 ztest_unit_test(test_mkdir),
-			 ztest_unit_test(test_opendir),
-			 ztest_unit_test(test_closedir),
-			 ztest_unit_test(test_opendir_closedir),
-			 ztest_unit_test(test_lsdir),
-			 ztest_unit_test(test_file_open),
-			 ztest_unit_test(test_file_write),
-			 ztest_unit_test(test_file_read),
-			 ztest_unit_test(test_file_seek),
-			 ztest_unit_test(test_file_truncate),
-			 ztest_unit_test(test_file_close),
-			 ztest_unit_test(test_file_sync),
-			 ztest_unit_test(test_file_rename),
-			 ztest_unit_test(test_file_stat),
-			 ztest_unit_test(test_file_unlink),
-			 ztest_unit_test(test_unmount),
-			 ztest_unit_test_setup_teardown(test_mount_flags,
-							dummy_setup,
-							fs_teardown)
-			 );
-	ztest_run_test_suite(fat_fs_basic_test);
+	test_fs_register();
+	test_mount();
+	test_fs_file_t_init();
+	test_fs_dir_t_init();
+	test_file_statvfs();
+	test_mkdir();
+	test_opendir();
+	test_closedir();
+	test_opendir_closedir();
+	test_lsdir();
+	test_file_open();
+	test_file_write();
+	test_file_read();
+	test_file_seek();
+	test_file_truncate();
+	test_file_close();
+	test_file_sync();
+	test_file_rename();
+	test_file_stat();
+	test_file_unlink();
+	test_unmount();
+	test_mount_flags();
```

Comment:
This turns all the tests into a single test. Can you split it up?

Reply:
@yperess  https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/subsys/fs/fs_api/src/main.c#L62-L63 said that the order of test cases is critical, one case depend on the case before it.

Reply:
@yperess please review again.